### ISSUE TITLE
[BUG] Point-in-polygon Primitive Should be Consistent Over Boundary Condition

### ISSUE THREAD

**Issue by isVoid** [rapidsai/cuspatial#744]
**Describe the bug**
Consider two axis aligned rectangle:
```
[(-1.0, -1.0), (0.0, -1.0), (0.0, 1.0), (-1.0, 1.0)]
[(0.0, -1.0), (1.0, -1.0), (1.0, 1.0), (0.0, 1.0)]
```
and a test point `(0, 0)`.

![Untitled Diagram (1)](https://user-images.githubusercontent.com/13521008/196828503-2ef2768a-af5d-4a73-a251-0014e2f6fd33.jpg)

This point is on the edge of these two rectangles. Is this point in the rectangle? The answer should only be always yes, or should be always no for both cases. Not yes for one and no for another.

> Not yes for one and no for another.
This is what's happening in current PIP kernel.

**Steps/Code to reproduce bug**
See test: https://github.com/rapidsai/cuspatial/blob/930a2f33953ce1abe8f0680ebbf418675db4b362/cpp/tests/experimental/spatial/point_in_polygon_test.cu#L157

**Expected behavior**
Result should either be `b0000` or `b1111`. Better still, user definable on boundary conditions.

**Comment by thomcom** [comment]
I think that #750 fixes this.

**Comment by harrism** [comment]
No, the crossingsMultiply algorithm is fundamentally inconsistent. Working on it.
